### PR TITLE
media: prevent Media nodes from being shared

### PR DESCRIPTION
**Pull request by cboesch-gpsw**

### CODE REVIEW

File: libnodegl/node_media.c [hunk 1]
```diff
@@ -213,6 +213,20 @@ static int media_init(struct ngl_node *node)
...
 }
 
+static int media_prepare(struct ngl_node *node)
+{
+    struct media_priv *s = node->priv_data;
+    if (s->nb_parents++) {
+        /*
+         * On Android, the frame can only be uploaded once, which discards
+         * them, so we can't have 2 Textures sharing the same Media.
+         */
```
Comment (mbouron-gpsw) [inline libnodegl/node_media.c:223]: I would say something like this insteaf: On Android, the frame can only be uploaded once and each subsequent upload will be a noop which results in an empty texture. This limitation prevents us from sharing the Media node.

This comment can also be put in the commit description.

File: libnodegl/node_media.c [hunk 2]
```diff
@@ -213,6 +213,20 @@ static int media_init(struct ngl_node *node)
...
+        LOG(ERROR, "Media nodes can not be shared, share the Texture instead");
```
Comment (mbouron-gpsw) [inline libnodegl/node_media.c:224]: Media nodes can not be shared, the Texture should be shared instead.

**Review by mbouron-gpsw — APPROVED** [review]
Other than my remarks, LGTM.

File: libnodegl/node_media.c (hunk 2 above)
Comment (cboesch-gpsw) [reply libnodegl/node_media.c]: Message adjusted

**Pull request merged**